releasenotes/notes/runtime-security-config-file-handling-a9508fda817b9b30.yaml

```diff
@@ -0,0 +1,6 @@
+---
+enhancements:
+  - |
+    The security agent will fail quickly if the first of either the default, or if they exist, user-provided override config files are inaccessible.
```

Comment:
```suggestion
    The security agent will fail immediately if the first of either the default, or if they exist, user-provided override config files are inaccessible.
```

Reply:
I'm having a hard time understanding this:
"if the first of either the default, or if they exist, user-provided override config files are inaccessible."

Reply:
Is this clearer?
```
The security agent will fail immediately if the first of either the default config files or user-provided override config files are inaccessible.

```

Reply:
Would taking out "the first of" affect the meaning? That part is kind of confusing. So:

```
The security agent will fail immediately if either the default config files or user-provided override config files are inaccessible.
```

Reply:
The the default config files are 2 files, and the user provided files are 0+. The security agent fails if the first of either of these sets are inaccessible.

Reply:
Thanks, that's really helpful. I think using some of that will make it clearer for the user. Does this make sense and is it correct? 

> There are two default config files. Optionally, you can also provide override config files. The security agent fails if the first config file of either of these sets are inaccessible. In the default case, there is the datadog.yaml and security-agent.yaml, and in that order. If datadog.yaml is inaccessible, the security agent fails immediately. If you provide overrides, like foo.yaml and bar.yaml, the security agent only fails immediately if foo.yaml is inaccessible.

Reply:
how about this? i wanted to highlight the change

```
There are two default config files. Optionally, you can provide override config files. 

The change in this release is that the security agent now fails if the first config file of either of these sets are inaccessible. In the default case, the config files are datadog.yaml and security-agent.yaml, and in that order. If datadog.yaml is inaccessible, the security agent fails immediately. If you provide overrides, like foo.yaml and bar.yaml, the security agent fails immediately if foo.yaml is inaccessible. 

In both sets, if any following config files are missing, the security agent continues to attempt to start up, with a log message about an inaccessible config file. 

Previously, the security-agent would continue attempting to start up if the first config file was inaccessible.  
```

Reply:
That looks good!